live-build/hooks/600-no-debian.binary

```diff
@@ -12,6 +12,27 @@ PREFIX=binary/boot/filesystem.dir
  chroot . dpkg -l > usr/share/snappy/dpkg.list
 )
 
+# generate dpkg.yaml file needed for OSS compliance
+(cd $PREFIX
+{
+  # fill in ppa information in yaml file
+  printf 'package-repositories:\n'
+  find etc/apt/ -name \*.list | while IFS= read -r APT; do
+    grep -o "^deb http://ppa.launchpad.net/[a-z0-9\.\+\-]\+/[a-z0-9\.\+\-]\+/[a-z0-9\.\+\-]\+" "$APT" | while read -r ENTRY ; do
+      USER=$(echo "$ENTRY" | cut -d/ -f4)
+      PPA=$(echo "$ENTRY" | cut -d/ -f5)
+      DISTRO=$(echo "$ENTRY" | cut -d/ -f6)
+      printf -- '- type: apt\n'
+      echo '  ppa: '"$USER/$DISTRO/$PPA"
+    done
+  done
+
+  # fill in yaml section with all installed packages
+  printf 'packages:\n'
+  chroot . dpkg-query -W | awk '{printf "- "}{printf $1}{printf "="}{print $2}'
```

Comment:
Thanks for this change - it seems with: `dpkg-query -W --showformat '- ${Package}=${Version}\n'` you can avoid using awk entirely, WDYT?

Reply:
Just note that it needs to be `${binary:Package}` rather than `${Package}` in order to preserve multiarch metadata correctly.  (I got this wrong in my first comment and had to edit it ...)

Reply:
Indeed, let me resolve this one so that only the correct suggestion is there. Thank you for spotting this!